docs/querying/nested-columns.md

```diff
@@ -0,0 +1,595 @@
+            "type": "json",
+            "name": "productDetails"
+          }
+        ]
+      },
+      "transformSpec": {
+        "transforms":[
+            { "type":"expression", "name":"firstName", "expression":"json_value(shipTo, '$.firstName')"},
+            { "type":"expression", "name":"lastName", "expression":"json_value(shipTo, '$.lastName')"},
+            { "type":"expression", "name":"address", "expression":"json_query(shipTo, '$.address')"},
+            { "type":"expression", "name":"productDetails", "expression":"json_object('product', product, 'details', details, 'department', department)"}
+        ]
+      }
+    },
+    "tuningConfig": {
+      "type": "index_parallel",
+      "partitionsSpec": {
+        "type": "dynamic"
+      }
+    }
+  }
+}
+```
+
+## SQL-based ingestion
+
+To ingest nested data using multi-stage query architecture, specify `COMPLEX<json>` as the value for `type` when you define the row signature&mdash;`shipTo` and `details` in the following example ingestion spec:
+
+![SQL-based ingestion](../assets/nested-msq-ingestion.png)
+
+```sql
+REPLACE INTO msq_nested_data_example OVERWRITE ALL
+SELECT
+  TIME_PARSE("time") as __time,
+  product,
+  department,
+  shipTo,
+  details
+FROM (
+  SELECT * FROM
+  TABLE(
+    EXTERN(
+      '{"type":"http","uris":["https://static.imply.io/data/nested_example_data.json"]}',
+      '{"type":"json"}',
+      '[{"name":"time","type":"string"},{"name":"product","type":"string"},{"name":"department","type":"string"},{"name":"shipTo","type":"COMPLEX<json>"},{"name":"details","type":"COMPLEX<json>"}]'
+    )
+  )
+)
+PARTITIONED BY ALL
+```
+
+### Transform data during SQL-based ingestion
+
+You can use the [JSON nested columns functions](./sql-json-functions.md) to transform JSON data in your ingestion query.
+
+For example, the following ingestion query is the SQL-based version of the [previous batch example](#transform-data-during-batch-ingestion)&mdash;it extracts `firstName`, `lastName`, and `address` from `shipTo` and creates a composite JSON object containing `product`, `details`, and `department`.
+
+![SQL-based ingestion](../assets/nested-msq-ingestion-transform.png)
+
+```sql
+REPLACE INTO msq_nested_data_transform_example OVERWRITE ALL
+SELECT
+  TIME_PARSE("time") as __time,
+  JSON_VALUE(shipTo, '$.firstName') as firstName,
+  JSON_VALUE(shipTo, '$.lastName') as lastName,
+  JSON_QUERY(shipTo, '$.address') as address,
+  JSON_OBJECT('product':product,'details':details, 'department':department) as productDetails
+FROM (
+  SELECT * FROM
+  TABLE(
+    EXTERN(
+      '{"type":"http","uris":["https://static.imply.io/data/nested_example_data.json"]}',
+      '{"type":"json"}',
+      '[{"name":"time","type":"string"},{"name":"product","type":"string"},{"name":"department","type":"string"},{"name":"shipTo","type":"COMPLEX<json>"},{"name":"details","type":"COMPLEX<json>"}]'
+    )
+  )
+)
+PARTITIONED BY ALL
+```
+
+## Ingest deserialized JSON as COMPLEX\<json>
```

Comment:
i think 'JSON string' or 'stringified' might be the more appropriate way to describe this data?

I also wonder if it is worth calling out that this is required for 'text' input formats such as TSV